t/t5538-push-shallow.sh

```diff
@@ -124,4 +124,21 @@ EOF
 	git cat-file blob $(echo 1|git hash-object --stdin) >/dev/null
 	)
 '
+
+test_expect_success 'push new commit from shallow clone to origin is efficient' '
+	git init origin &&
+	echo a >origin/a &&
+	git -C origin add a &&
+	git -C origin commit -m "base" &&
+	echo b >origin/b &&
+	git -C origin add b &&
+	git -C origin commit -m "tip" &&
+
+	git clone --depth=1 "file://$(pwd)/origin" client &&
+	git -C client checkout -b topic &&
+	git -C client commit --allow-empty -m "empty" &&
+	GIT_PROGRESS_DELAY=0 git -C client push --progress origin topic 2>err &&
+	test_grep "Enumerating objects: 1, done." err
```

Comment:
I _love_ how straight-forward this test case reads: an empty commit must not add a new blob nor tree, and therefore all that needs to be pushed is the new commit object, which I would not expect to delta well against any other commit at all.

Technically, this improvement has little to do with _path-walking_, I guess, the difference is all in the way boundary commits are considered, right?